client/state/sites/reducer.js

```diff
@@ -74,9 +75,12 @@ export function items( state = {}, action ) {
 			// Normalize incoming site(s) to array
 			const sites = action.site ? [ action.site ] : action.sites;
 
+			if ( SITES_UPDATE === action.type && state === null ) {
+				return state;
+			}
 			// SITES_RECEIVE occurs when we receive the entire set of user
 			// sites (replace existing state). Otherwise merge into state.
-			const initialNextState = SITES_RECEIVE === action.type ? {} : state;
+			const initialNextState = SITES_RECEIVE === action.type || state === null ? {} : state;
```

Comment:
(non-blocking, optional) Instead of adding `action.type` checks here, maybe it would be more readable if we moved them to appropriate `case` block (e.g. `case SITE_RECEIVE:`) and factor out the common functionality into helper.